pool/src/lib.rs

```diff
@@ -688,6 +690,46 @@ impl LendingPoolTrait for LendingPool {
         Ok(())
     }
 
+    fn set_as_collateral(
+        env: Env,
+        who: Address,
+        asset: Address,
+        use_as_collateral: bool,
+    ) -> Result<(), Error> {
+        who.require_auth();
+        let mut user_config = read_user_config(&env, who.clone())?;
+        let reserve_index = read_reserve(&env, asset)?.get_id();
+        if user_config.is_borrowing_any() {
+            if use_as_collateral {
+                assert_with_error!(
+                    &env,
+                    !user_config.is_borrowing(&env, reserve_index),
+                    Error::MustNotHaveDebt
+                );
+            } else if !user_config.is_borrowing(&env, reserve_index) {
+                user_config.set_using_as_collateral(&env, reserve_index, use_as_collateral);
```

Comment:
Let's consider rewriting it to not call set_using_as_collateral twice. Maybe match will help.
<img width="622" alt="image" src="https://github.com/eq-lab/Slender/assets/12034486/5c3fd374-9de4-4951-b863-4a60e6e5b4e2">